src/main/python/onnx_systemds/operator_gen.py

```diff
@@ -0,0 +1,476 @@
+        operator=operator[node.op_type],
+        doc_string=node.doc_string
+    )
+    return GeneratedScriptPart(node_render)
+
+
+def gen_1input_1output_mat_operator(env: jinja2.environment.Environment, graph: onnx.GraphProto,
+                                    node: onnx.NodeProto) -> GeneratedScriptPart:
+    """
+    Generates simple operators like 'y = -x' which have one input and one output.
+    :param env:  Jinja environment to load the template files
+    :param graph: the onnx-graph for which the script shall be generated
+    :param node: the onnx-node for which the script shall be generated
+    :return: The generated script part
+    """
+    template_for_operator = {
+        "Neg": "neg.dml.jinja",
+    }
+
+    operator_template = env.get_template("operators/" + template_for_operator[node.op_type])
+
+    if len(node.attribute) != 0:
+        raise Exception("attributes not supported for operator")
+
+    if len(list(node.input)) != 1 or len(list(node.output)) != 1:
+        raise Exception("Operator needs 1 input and 1 output")
+
+    node_render = operator_template.render(
+        input=list(node.input)[0],
+        output=list(node.output)[0],
+        doc_string=node.doc_string
+    )
+    return GeneratedScriptPart(dml_script=node_render)
+
+
+def gen_dropout_call(env: jinja2.environment.Environment, graph: onnx.GraphProto,
+                     node: onnx.NodeProto) -> GeneratedScriptPart:
+    operator_template = env.get_template("operators/" + "function_call.dml.jinja")
+    import_template = env.get_template("module_import.dml.jinja")
+
+    function_namespace = "dropout_layer"
+    function_name = "forward"
+    path = "/nn/layers/dropout.dml"
+
+    #  * Inputs:
+    #    *  - X: Inputs, of shape (any, any).
+    #    *  - p: Probability of keeping a neuron output.
+    #    *  - seed: [Optional: -1] Random number generator seed to allow for
+    #    *      deterministic evaluation.  Set to -1 for a random seed.
+    # * Outputs:
+    #    *  - out: Outputs, of same shape as `X`.
+    #    *  - mask: Dropout mask used to compute the output.
+
+    X = list(node.input)[0]
+    p = 0.5
+    seed = -1
+    if len(list(node.attribute)) > 0:
+        attributes = list(node.attribute)
+        if attributes[0].name != "ratio" or len(attributes) > 1:
+            raise Exception("Error in generating dropout call invalid attributes" + str(attributes))
+        p = attributes[0].f
+
+    import_render = import_template.render(
+        path=path,
+        name=function_namespace
+    )
+
+    node_render = operator_template.render(
+        function_namespace=function_namespace,
+        function=function_name,
+        arguments=[X, p, seed],
+        outputs=list(node.output),
+        doc_string=node.doc_string
+    )
+    return GeneratedScriptPart(imports=[import_render], dml_script=node_render)
+
+
+def gen_maxpool_call(env: jinja2.environment.Environment, graph: onnx.GraphProto,
```

Comment:
gen conv and gen maxpool, contain much of the same code, maybe some of it could be alleviated using some private helper functions?